acquire/acquire.py

```diff
@@ -1779,21 +1780,34 @@ def print_acquire_warning(target: Target) -> None:
 
 
 def _get_modules_for_profile(
+    target: Target,
     profile_name: str,
-    operating_system: str,
     profiles: dict[str, dict[str, list[type[Module]]]],
     err_msg: str,
 ) -> dict[str, type[Module]]:
-    modules_selected = {}
+    if profile_name == "none":
+        return {}
 
-    if profile_name != "none":
-        if (profile := profiles.get(profile_name, {}).get(operating_system)) is not None:
-            for mod in profile:
-                modules_selected[mod.__modname__] = mod
-        else:
-            log.error(err_msg, operating_system, profile_name)
+    if (profile_os := profiles.get(profile_name)) is None:
+        log.error("No profile found named %s", profile_name)
+        return {}
+
+    if (profile := profile_os.get(target.os)) is None:
+        for os in target.os_tree():
+            if profile := profile_os.get(os):
+                log.info("No profile for %s, using %s instead", target.os, os)
```

Comment:
```suggestion
                log.info("No collection set for OS %r with profile %r, using the one for OS %r instead", target.os, profile_name, os)
```
Made it more like the other log messages. Also used `%r` for the free quotes.